cmd/apiserver-boot/boot/run/local.go

```diff
@@ -178,45 +159,58 @@ func RunApiserver(stopCh chan struct{}) *exec.Cmd {
 	apiserverCmd := exec.Command(server,
 		flags...,
 	)
-	fmt.Printf("%s\n", strings.Join(apiserverCmd.Args, " "))
 	if printapiserver {
 		apiserverCmd.Stderr = os.Stderr
 		apiserverCmd.Stdout = os.Stdout
 	}
 
-	go func() {
-		err := apiserverCmd.Run()
-		if err != nil {
-			log.Printf("Failed to run apiserver %v", err)
-			stopCh <- struct{}{}
-		}
-	}()
+	go runCommon(apiserverCmd, ctx, cancel)
 
 	return apiserverCmd
 }
 
-func RunControllerManager(stopCh chan struct{}) *exec.Cmd {
+func RunControllerManager(ctx context.Context, cancel context.CancelFunc) *exec.Cmd {
 	if len(controllermanager) == 0 {
 		controllermanager = "bin/controller-manager"
 	}
+
 	controllerManagerCmd := exec.Command(controllermanager,
 		fmt.Sprintf("--kubeconfig=%s", config),
 	)
-	fmt.Printf("%s\n", strings.Join(controllerManagerCmd.Args, " "))
 	if printcontrollermanager {
 		controllerManagerCmd.Stderr = os.Stderr
 		controllerManagerCmd.Stdout = os.Stdout
 	}
 
+	go runCommon(controllerManagerCmd, ctx, cancel)
+
+	return controllerManagerCmd
+}
+
+// run a command via goroutine
+func runCommon(cmd *exec.Cmd, ctx context.Context, cancel context.CancelFunc) {
+	stopCh := make(chan error)
+	cmdName := cmd.Args[0]
+
+	fmt.Printf("%s\n", strings.Join(cmd.Args, " "))
 	go func() {
-		err := controllerManagerCmd.Run()
+		err := cmd.Run()
 		if err != nil {
-			log.Printf("Failed to run controller-manager %v", err)
-			stopCh <- struct{}{}
+			log.Printf("Failed to run %s, error: %v\n", cmdName, err)
+			stopCh <- err
```

Comment:
You can save this line and the else and just `stopCh<-err` - it'll be nil in the else case either way.

Reply:
ok, I will update this.